pkg/shortnames/shortnames_test.go

```diff
@@ -0,0 +1,527 @@
+		assert.Equal(t, test.repo, normalized)
+		assert.Equal(t, test.isTagged, isTagged)
+		assert.Equal(t, test.isDigested, isDigested)
+		if isTagged {
+			normalized = normalized + ":" + tag
+		} else if isDigested {
+			normalized = normalized + "@" + digest.String()
+		}
+		assert.Equal(t, test.input, normalized)
+	}
+}
+
+func TestResolve(t *testing.T) {
+	tmp, err := ioutil.TempFile("", "aliases.conf")
+	require.NoError(t, err)
+	defer os.Remove(tmp.Name())
+
+	sys := &types.SystemContext{
+		SystemRegistriesConfPath:    "testdata/aliases.conf",
+		SystemRegistriesConfDirPath: "testdata/this-does-not-exist",
+		UserShortNameAliasConfPath:  tmp.Name(),
+	}
+
+	_, err = sysregistriesv2.TryUpdatingCache(sys)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name, value string
+	}{
+		{"docker", "docker.io/library/foo:latest"},
+		{"docker:tag", "docker.io/library/foo:tag"},
+		{
+			"docker@sha256:d366a4665ab44f0648d7a00ae3fae139d55e32f9712c67accd604bb55df9d05a",
+			"docker.io/library/foo@sha256:d366a4665ab44f0648d7a00ae3fae139d55e32f9712c67accd604bb55df9d05a",
+		},
+		{"quay/foo", "quay.io/library/foo:latest"},
+		{"quay/foo:tag", "quay.io/library/foo:tag"},
+		{
+			"quay/foo@sha256:d366a4665ab44f0648d7a00ae3fae139d55e32f9712c67accd604bb55df9d05a",
+			"quay.io/library/foo@sha256:d366a4665ab44f0648d7a00ae3fae139d55e32f9712c67accd604bb55df9d05a",
+		},
+		{"example", "example.com/library/foo:latest"},
+		{"example:tag", "example.com/library/foo:tag"},
+		{
+			"example@sha256:d366a4665ab44f0648d7a00ae3fae139d55e32f9712c67accd604bb55df9d05a",
+			"example.com/library/foo@sha256:d366a4665ab44f0648d7a00ae3fae139d55e32f9712c67accd604bb55df9d05a",
+		},
+	}
+
+	// All of them should resolve correctly.
+	for _, test := range tests {
+		resolved, err := Resolve(sys, test.name)
+		require.NoError(t, err, "%v", test)
+		require.NotNil(t, resolved)
+		require.Len(t, resolved.PullCandidates, 1)
+		assert.Equal(t, test.value, resolved.PullCandidates[0].Value.String())
+		assert.False(t, resolved.PullCandidates[0].record)
+	}
+
+	// Non-existent should return an empty slice as no search registries
+	// are configured in the config.
+	resolved, err := Resolve(sys, "dontexist")
+	require.NoError(t, err)
+	require.NotNil(t, resolved)
+	require.Len(t, resolved.PullCandidates, 0)
+
+	// An empty name is not valid.
+	resolved, err = Resolve(sys, "")
+	require.Error(t, err)
+	require.Nil(t, resolved)
+
+	// Invalid input.
+	resolved, err = Resolve(sys, "Invalid#$")
+	require.Error(t, err)
+	require.Nil(t, resolved)
+
+	// Fully-qualified input will be returned as is.
+	resolved, err = Resolve(sys, "quay.io/repo/fedora")
+	require.NoError(t, err)
+	require.NotNil(t, resolved)
+	require.Len(t, resolved.PullCandidates, 1)
+	assert.Equal(t, "quay.io/repo/fedora:latest", resolved.PullCandidates[0].Value.String())
+	assert.False(t, resolved.PullCandidates[0].record)
+}
+
+func toNamed(input string, trim bool) (reference.Named, error) {
```

Comment:
This could be `toNamed(t *testing.T, …)` and abort on failure, saving extra checks in callers.

Reply:
done